src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc

```diff
@@ -385,19 +387,9 @@ BtreeLBAManager::_alloc_extent(
 	    state.ret = iter;
 	  });
 	});
-    }).si_then([c, actual_addr, addr, intermediate_base,
-		intermediate_length](auto &&state) {
-      auto ret_pin = state.ret->get_pin(c);
-      if (actual_addr != P_ADDR_NULL) {
-	ceph_assert(addr.is_laddr());
-	ret_pin->set_paddr(actual_addr);
-	ret_pin->set_intermediate_base(intermediate_base);
-	ret_pin->set_intermediate_length(intermediate_length);
-      } else {
-	ceph_assert(addr.is_paddr());
-      }
-      return alloc_extent_iertr::make_ready_future<LBAMappingRef>(
-	std::move(ret_pin));
```

Comment:
The standalong commit ""better code readability" isn't correct without the commit "drop TransactionManager::inc_ref()" around this logic.

How about merging the two commits into a more complete one?

Reply:
ok